windows/installer.nsi

```diff
@@ -365,45 +379,81 @@ Function createdesktopshortcut
    CreateShortCut  "$DESKTOP\${APP_NAME}.lnk" "$INSTDIR\${APP_EXE}"
 FunctionEnd
 
-Function ASIOCheckInstalled
-
-    ; insert ASIO install page if no ASIO driver was found
-    ClearErrors
-    EnumRegKey $0 HKLM "SOFTWARE\ASIO" 0
-
-    IfErrors 0 ASIOExists
-        !insertmacro MUI_HEADER_TEXT "$(ASIO_DRIVER_HEADER)" "$(ASIO_DRIVER_SUB)"
-        nsDialogs::Create 1018
-        Pop $Dialog
-        ${If} $Dialog == error
-            Abort
-        ${Endif}
-
-        ${NSD_CreateLabel} 0 0 100% 20u "$(ASIO_DRIVER_EXPLAIN)"
-        Pop $Label
-        ${NSD_CreateButton} 0 21u 100% 15u "$(ASIO_DRIVER_MORE_INFO)"
-        Pop $Button
-        ${NSD_OnClick} $Button OpenASIOHelpPage
-
-        nsDialogs::Show
-
-    ASIOExists:
-
-FunctionEnd
-
-Function OpenASIOHelpPage
-    ExecShell "open" "$(ASIO_DRIVER_MORE_INFO_URL)"
-FunctionEnd
-
-Function ExitASIOInstalled
-    ClearErrors
-    EnumRegKey $0 HKLM "SOFTWARE\ASIO" 0
-    IfErrors 0 SkipMessage
-        MessageBox MB_YESNO|MB_ICONEXCLAMATION "$(ASIO_EXIT_NO_DRIVER)" /sd IDNO IDYES SkipMessage
-            Abort
-   SkipMessage:
-
-FunctionEnd
+; Define functions for JACK when using the jackonwindows buildoption, else use default ASIO functions
+!if ${BUILD_OPTION} == "jackonwindows"
+    Function JACKCheckInstalled
+
+        ; insert JACK install page if no JACK driver was found
+        ClearErrors
+        EnumRegKey $0 HKLM "SOFTWARE\JACK" 0
+
+        IfErrors 0 JACKExists
+            !insertmacro MUI_HEADER_TEXT "$(JACK_DRIVER_HEADER)" "$(JACK_DRIVER_SUB)"
+            nsDialogs::Create 1018
+            Pop $Dialog
+            ${If} $Dialog == error
+                Abort
+            ${Endif}
+
+            ${NSD_CreateLabel} 0 0 100% 40u "$(JACK_DRIVER_EXPLAIN)"
+            Pop $Label
+
+            nsDialogs::Show
+
+        JACKExists:
+
+    FunctionEnd
+
+    Function ExitJACKInstalled
+        ClearErrors
+        EnumRegKey $0 HKLM "SOFTWARE\JACK" 0
+        IfErrors 0 SkipMessage
+            MessageBox MB_YESNO|MB_ICONEXCLAMATION "$(JACK_EXIT_NO_DRIVER)" /sd IDNO IDYES SkipMessage
```

Comment:
I hope SkipMessage jumps to line 413 not 453. It doesn't matter here anyway.

What about an !include which only gets included during compile time depending on which version we're building?

Reply:
The page and functions are not "seen" by the code, they are removed/ignored in the compile step. For this reason I also had to make sure variables that were not used for the "jackonwindows" option were not defined, as nsis would complain about being there and not used.